src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11652,18 +11662,21 @@ int Assembler::vex_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexS
             (!_legacy_mode_vl) ||
             (attributes->is_legacy_mode())),"XMM register should be 0-15");
     // Instruction with legacy_mode true should have dst, nds and src < 15
-    assert(((dst_enc < 16 && nds_enc < 16 && src_enc < 16) || (!attributes->is_legacy_mode())),"XMM register should be 0-15");
+    assert(((dst_enc < 16 && nds_enc < 16 && src_enc < 16) || has_gpr || (!attributes->is_legacy_mode())),"XMM register should be 0-15");
```

Comment:
This could then  be changed to:
assert(((!is_extended) || (!attributes->is_legacy_mode())),"XMM register should be 0-15");

Reply:
Thanks, done.